src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,617 @@
+            $this->addLineStmt($this->createMethod($method, $argValue, $argKey));
+        }
+    }
+
+    private function createMethod(string $method, $value, $keyAsFirstArgument = false): string
+    {
+        $value = $this->toString($value);
+        list($method, $value) = $this->adaptMethodIfNecessary($method, $value);
+
+        return $keyAsFirstArgument
+            ? $this->tab().'->'.$method.'('.$this->createStringArgument($keyAsFirstArgument).', '.$value.')'
+            : $this->tab().'->'.$method.'('.$value.')';
+    }
+
+    private function createMethodFromSequentialArray(string $method, array $arrayValues): string
+    {
+        $arguments = "[\n";
+        foreach ($arrayValues[$method] as $subKey => $subValue) {
+            $arguments .= $this->tab(3).$this->toString($subValue, true).",\n";
+        }
+        list($method) = $this->adaptMethodIfNecessary($method, $arrayValues);
+
+        return $this->tab().'->'.$method.'('.$arguments.$this->tab(2).'])';
+    }
+
+    private function createDecorateMethod(array $value): string
+    {
+        $innerName = isset($value['decoration_inner_name'])
+            ? ', '.$this->toString($value['decoration_inner_name'])
+            : ', null';
+
+        $priority = isset($value['decoration_priority'])
+            ? ', '.$this->toString($value['decoration_priority'])
+            : ', 0';
+
+        $invalid = isset($value['decoration_on_invalid'])
+            ? $value['decoration_on_invalid']
+            : null;
+
+        if ($invalid) {
+            $this->addUseStatementIfNecessary('Symfony\Component\DependencyInjection\ContainerInterface');
+            $invalid = 'exception' === $invalid
+                ? ', ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE'
+                : ', ContainerInterface::IGNORE_ON_INVALID_REFERENCE';
+        }
+        // Don't write the next arguments if they are null.
+        if (null === $invalid && ', 0' === $priority) {
+            $priority = null;
+            $innerName = ', null' === $innerName ? null : $innerName;
+        }
+
+        return $this->tab().'->decorate('.$this->createStringArgument($value['decorates']).$innerName.$priority.$invalid.')';
+    }
+
+    private function createCallMethod(array $values): string
+    {
+        $firstKey = array_key_first($values);
+        $firstValue = $values[$firstKey];
+
+        $value = $this->tab().'->call('.$this->createStringArgument($firstValue).', ['.$this->toString($values[1]).']';
+        // ->call() can have 3 arguments.
+        if (isset($values[2]) && \is_bool($values[2])) {
+            return $values[2] ? $value.', true)' : $value.', false)';
+        }
+
+        return $value.')';
+    }
+
+    private function createTagMethod($value): string
+    {
+        if (!\is_array($value)) {
+            return $this->tab().'->tag('.$this->createStringArgument($value).')';
+        }
+
+        $name = $this->createStringArgument($value['name']);
```

Comment:
I offer more short and more  consistent because we don't use array() function in other cases and we have spécific methods for creating beautiful arrays :